### ISSUE TITLE
🐛 alertdialog should be permitted on `dialog` elements

### ISSUE THREAD

**Issue by lifeiscontent** [biomejs/biome#3964]
### Environment information

```block
as outlined here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#technical_summary

the `alertdialog` should be a permitted role for `lint/a11y/useSemanticElements` on `<dialog />` elements.

CLI:
  Version:                      1.9.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.12.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.10.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. I defined a `dialog` element with the `role` of `alertdialog`

### Expected result

I should not see an error for `lint/a11y/useSemanticElements`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Could you add a playground example for reproduction?

**Comment by lifeiscontent** [comment]
@Conaclos ah, you know what, I'm in the middle of porting the codebase from ESLint to Biome, it was ESLint, sorry for the false report.
